app/src/main/kotlin/me/amrbashir/hijriwidget/preference_activity/LauncherIcon.kt

```diff
@@ -3,33 +3,41 @@ package me.amrbashir.hijriwidget.preference_activity
 import android.content.ComponentName
 import android.content.Context
 import android.content.pm.PackageManager
+import android.os.Build
 import me.amrbashir.hijriwidget.BuildConfig
 import me.amrbashir.hijriwidget.HijriDate
 import me.amrbashir.hijriwidget.PreferencesManager
 
 fun changeLauncherIcon(context: Context, prefsManager: PreferencesManager) {
-    val today = HijriDate.todayNumber(prefsManager)
+    val actions = buildList {
+        if (!BuildConfig.DEBUG) {
+            val cls = "me.amrbashir.hijriwidget.preference_activity.PreferenceActivity"
+            val mainActivity = ComponentName(context, cls)
+            add(mainActivity to PackageManager.COMPONENT_ENABLED_STATE_DISABLED)
+        }
 
-    val packageManager = context.packageManager
-
-    if (!BuildConfig.DEBUG) {
-        val cls = "me.amrbashir.hijriwidget.preference_activity.PreferenceActivity"
-        val mainActivity = ComponentName(context, cls)
-        packageManager.setComponentEnabledSetting(
-            mainActivity,
-            PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
-            PackageManager.DONT_KILL_APP
-        )
+        val today = HijriDate.todayNumber(prefsManager)
+        for (day in 1..30) {
+            val cls = "me.amrbashir.hijriwidget.preference_activity.Calendar_$day"
+            val dayActivity = ComponentName(context, cls)
+            val newState = if (today == day) PackageManager.COMPONENT_ENABLED_STATE_ENABLED
+            else PackageManager.COMPONENT_ENABLED_STATE_DISABLED
+            add(dayActivity to newState)
+        }
     }
 
-    for (day in 1..30) {
-        val cls = "me.amrbashir.hijriwidget.preference_activity.Calendar_$day"
-        val dayActivity = ComponentName(context, cls)
-        packageManager.setComponentEnabledSetting(
-            dayActivity,
-            if (today == day) PackageManager.COMPONENT_ENABLED_STATE_ENABLED else PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
-            PackageManager.DONT_KILL_APP
+    val packageManager = context.packageManager
+    val flags = PackageManager.DONT_KILL_APP
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        packageManager.setComponentEnabledSettings(
+            actions.map { (componentName, newState) ->
+                PackageManager.ComponentEnabledSetting(componentName, newState, flags)
+            }
         )
```

Comment:
Small nitpick, I feel this makes it a bit more readable and avoids an extra nesting level
```suggestion
        val actions = actions.map { (componentName, newState) ->
            PackageManager.ComponentEnabledSetting(componentName, newState, flags)
        }
        packageManager.setComponentEnabledSettings(actions)
```